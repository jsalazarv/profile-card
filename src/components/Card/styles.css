.card {
    @apply overflow-auto;
    @apply flex;
    @apply flex-col;
}

.card .card-header {
    @apply w-full;
}
