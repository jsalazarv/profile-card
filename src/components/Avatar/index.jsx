import './styles.css';
import { useMemo } from 'react';

export const Avatar = ({
  className = '',
  children,
  size = 'small',
  type = 'circle',
  color = 'bg-primary',
}) => {
  const sizes = {
    small: 'w-6 h-6',
    medium: 'w-16 h-16',
    large: 'w-32 h-32',
  };

  const types = {
    circle: 'rounded-full',
    rounded: 'rounded-lg',
    tile: 'rounded-none',
  };

  const avatarSize = useMemo(() => sizes[size], [size]);
  const avatarType = useMemo(() => types[type], [type]);

  return (
    <div
      className={`overflow-auto ${avatarSize} ${avatarType} ${color} ${className}`}>
      {children}
    </div>
  );
};
