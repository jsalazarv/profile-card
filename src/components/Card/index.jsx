import './styles.css';
import { useMemo } from 'react';

export const Card = ({
  className = '',
  children,
  type = 'tile',
  outlined = '',
}) => {
  const types = {
    rounded: 'rounded-lg',
    tile: 'rounded-none',
  };
  const outline = 'border-solid border';
  const cardType = useMemo(() => types[type], [type]);

  return (
    <div className={`card ${cardType} ${outlined && outline} ${className}`}>
      {children}
    </div>
  );
};
