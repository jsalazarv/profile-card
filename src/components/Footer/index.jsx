import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHeart } from '@fortawesome/free-solid-svg-icons';
import './styles.css';

export const Footer = ({ phrase = 'Designed with' }) => {
  return (
    <>
      <footer className="footer">
        <p>
          {phrase}
          <span>
            <FontAwesomeIcon icon={faHeart} />
          </span>
        </p>
      </footer>
    </>
  );
};
