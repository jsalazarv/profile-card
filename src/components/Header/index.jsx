import './styles.css';
import { ThemeToggle } from '../ThemeToggle/index.jsx';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faGithub } from '@fortawesome/free-brands-svg-icons';

export const Header = ({}) => {
  return (
    <header className="header">
      <nav className="header__navbar" aria-label="Top">
        <div className="navbar__content">
          <div className="logo__container">
            <a href="https://jsalazarv.dev/" target="_blank">
              <h2>[Js]</h2>
            </a>
          </div>
          <div className="items__container">
            <a href="https://github.com/jsalazarv/profile-card" target="_blank">
              <FontAwesomeIcon
                className="text-2xl border-black dark:text-amber-500 dark:text-3xl transition-all duration-1000"
                icon={faGithub}
              />
            </a>
            <ThemeToggle />
          </div>
        </div>
      </nav>
    </header>
  );
};
