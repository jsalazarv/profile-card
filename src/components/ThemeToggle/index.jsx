import './styles.css';
import useDarkMode from '../../hooks/useDarkMode.js';

export const ThemeToggle = ({ className = '' }) => {
  const [colorTheme, setTheme] = useDarkMode();

  const toggleTheme = () => {
    setTheme(colorTheme);
  };

  return (
    <input
      className={`toggle ${className}`}
      type="checkbox"
      onClick={toggleTheme}
      aria-label="toggleTheme"
    />
  );
};
