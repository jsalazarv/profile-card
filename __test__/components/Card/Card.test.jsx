import { render } from '@testing-library/react';
import { Card } from '../../../src/components/Card';

describe('[Card Component]', () => {
  describe('[Snapshots]', () => {
    test('It should represent the default values', () => {
      const { container } = render(<Card />);

      expect(container.firstChild).toHaveClass('card');
      expect(container.firstChild).toHaveClass('rounded-none');
      expect(container).toMatchSnapshot();
    });

    test('It should contain the class [rounded-lg] if the configured prop is [rounded]', () => {
      const cardType = 'rounded';
      const { container } = render(<Card type={cardType} />);

      expect(container.firstChild).toHaveClass('rounded-lg');
      expect(container.firstChild).not.toHaveClass('rounded-none');
      expect(container).toMatchSnapshot();
    });

    test('It should contain the class [rounded-none] if the configured prop is [tile]', () => {
      const cardType = 'tile';
      const { container } = render(<Card type={cardType} />);

      expect(container.firstChild).toHaveClass('rounded-none');
      expect(container.firstChild).not.toHaveClass('rounded-lg');
      expect(container).toMatchSnapshot();
    });

    test('It should contain the class [border-solid border] if the configured prop is [outlined]', () => {
      const { container } = render(<Card outlined />);

      expect(container.firstChild).toHaveClass('border-solid border');
      expect(container).toMatchSnapshot();
    });
  });
});
