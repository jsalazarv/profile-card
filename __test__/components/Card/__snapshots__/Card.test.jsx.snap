// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Card Component] [Snapshots] It should contain the class [border-solid border] if the configured prop is [outlined] 1`] = `
<div>
  <div
    class="card rounded-lg border-solid border "
  />
</div>
`;

exports[`[Card Component] [Snapshots] It should contain the class [rounded-lg] if the configured prop is [rounded] 1`] = `
<div>
  <div
    class="card rounded-lg  "
  />
</div>
`;

exports[`[Card Component] [Snapshots] It should contain the class [rounded-none] if the configured prop is [tile] 1`] = `
<div>
  <div
    class="card rounded-none  "
  />
</div>
`;

exports[`[Card Component] [Snapshots] It should represent the default values 1`] = `
<div>
  <div
    class="card rounded-lg  "
  />
</div>
`;

exports[`[Card Component] [Snapshots] Must not contain the class [border-solid border] if the [outlined] property is not set 1`] = `
<div>
  <div
    class="card rounded-lg  "
  />
</div>
`;
